src/foam/nanos/session/Session.js

```diff
@@ -352,6 +352,14 @@ List entries are of the form: 172.0.0.0/24 - this would restrict logins to the 1
 
         PM pm = PM.create(x, "Session", "applyTo", "create");
 
+        // create a temp session with the user/agent from the old session context to check if it was an anonymous session
+        // and set the wasAnonymous boolean accordingly
+        boolean wasAnonymous = false;
+        if ( subjectUser != null ) {
+          X oldX = (X) foam.util.Auth.sudo(x, subjectUser, subjectAgent);
+          wasAnonymous = auth.isAnonymous(oldX);
+        }
```

Comment:
How about: 
if ( subjectUser != null ) {
  ServiceProvider sp = (ServiceProvider) subjectUser.findSpid(x);
  if ( sp != null && 
        sp.getAnonymousUser() == subjectUser.getId() ) {
    wasAnonymous = true;
  }
}
Thinking about performance and calling sudo which itself calls applyTo.